### ISSUE TITLE
Cron-based backup of TRMM with backup rotation.

### ISSUE THREAD

**Issue by fts-tmassey** [amidaware/tacticalrmm#1056]
I created a very straightforward Bash script to generate TRMM backups via cron, as well as only keep a maximum number of backups. There's nothing very clever in this, but if I can save a few minutes for each new installation, then great.

https://github.com/fts-tmassey/tacticalrmm-cronbackup

**Comment by fts-tmassey** [comment]
Closing issue.

**Comment by dinger1986** [comment]
thanks

**Comment by fts-tmassey** [comment]
I went ahead and expanded the installation/configuration information in the readme:  https://github.com/fts-tmassey/tacticalrmm-cronbackup/blob/main/README.md

It's literally a single file, and it needs to be updated based on what user and path the admin used to install TRMM.  Given that TRMM doesn't specify that information, it could be *anything*.  Seeing as building a script to collect and manage that information properly would be longer than the script itself...  I'm going to leave it with manual installation.

Besides, if you can install and administer an RMM system that is actively under construction, I really hope you can handle copying a single file and updating a couple of variables...  :)  But I did expand on the necessary details, with suggested actions:  if you follow the doc, you should get a working process.

**Comment by dinger1986** [comment]
> I went ahead and expanded the installation/configuration information in the readme: https://github.com/fts-tmassey/tacticalrmm-cronbackup/blob/main/README.md
> 
Thanks!

> It's literally a single file, and it needs to be updated based on what user and path the admin used to install TRMM. Given that TRMM doesn't specify that information, it could be _anything_. Seeing as building a script to collect and manage that information properly would be longer than the script itself... I'm going to leave it with manual installation.
> 
Agreed but some wont know how to use it, if it was integrated into a script it would pull all the info, I can always look at that.

> Besides, if you can install and administer an RMM system that is actively under construction, I really hope you can handle copying a single file and updating a couple of variables... :) But I did expand on the necessary details, with suggested actions: if you follow the doc, you should get a working process.
>
Lol you would think everyone running these servers could it, scary sometimes!

**Comment by fts-tmassey** [comment]
I just updated my readme to also include how to get the backup.sh script in the first place.  In doing so, I put in a link to the TRMM backup doc, and noticed that the included wget (to get the script) points at the old repo.  You will likely want to update that for the new repo name.  (That's gonna be a lot of ongoing cleanup fun for the team...)

**Comment by dinger1986** [comment]
wget looks right to me, says amidaware?

**Comment by dinger1986** [comment]
![image](https://user-images.githubusercontent.com/7244447/162470994-73852967-5c6d-43dd-bce5-0279a82f25e0.png)

**Comment by fts-tmassey** [comment]
Not in the doc:  https://amidaware.github.io/tacticalrmm/backup/

**Comment by dinger1986** [comment]
eh ok, were using https://docs.tacticalrmm.com/, wonder wtf the other one is! We will get it sorted

**Comment by fts-tmassey** [comment]
Ahh, the sin of multiple copies...  :)  I've updated the readme to point to the docs URL instead.

**Comment by dinger1986** [comment]
yeah need to sort that out! Thanks for pointing it out

**Comment by silversword411** [comment]
Looks awesome to me! :)

https://github.com/amidaware/trmm-awesome/commit/a806d77fd4df253d22b666706e8ee1ad669212e9

**Comment by fts-tmassey** [comment]
While rooting around for something else, I found that there's a  backup.sh in the /rmm directory.  Is this now officially part of the install?  If so, two things:

1) You may want to update the backup documentation and document that, instead of having the user wget one ( https://docs.tacticalrmm.com/backup/ ), and 
2) I will update the backup script to use that path for the script:  that's one thing less for the user to configure.

Thanks for the info!

**Comment by dinger1986** [comment]
I'm sure it gets downloaded as the install script grabs the git repo, maybe even updates do as well?

**Comment by fts-tmassey** [comment]
I'm not worried about updates:  the backup.sh script takes care of itself.  I'm worried about why the doc says to manually download the script, when it's already in /rmm.  If its presence in /rmm is an accident, the doc makes sense, and I need to ask the user to do a few extra steps.  But if the presence of the script is both intentional and supported, I can remove all that stuff from the process of setting up the script.  And if that's the case, the doc should be updated to remove that extraneous stuff, too.

In other words:  do the developers "bless" the presence of backup.sh in /rmm?

It makes all the sense in the world for it to be there and to stay there.  All I need is someone to tell me that the *developers* agree with that - and if that's the case, it would be wise to update the doc to reflect this as well.

**Comment by dinger1986** [comment]
I guess all that's needed is comparing backup.sh with the current one in the repo and see if they are the same.

I'll check mine and confirm, saves the devs time unless they get back before then lol.

**Comment by fts-tmassey** [comment]
Again, the problem is not is the script there or is it right.  It *is* there, and it *is* right.  The question is:  is it *supported*?  Only the devs can answer that.  And why am I asking?  Because that directly conflicts with the documentation.  If the doc agreed, I wouldn't be asking right now -- but I wouldn't have done it the "wrong" way, either.

I'm not asking for a technical answer:  I *know* it will work today.  But is it *supported*?

**Comment by silversword411** [comment]
> I'm worried about why the doc says to manually download the script,

It says that....because.

So actually the entire /tacticalrmm repo is there as of your last git sync...and update.sh checks to make sure he's updated in case of user craziness...but depending on last git sync as part of update.sh, or something else docs officially just make you explicitly get the right...and latest when doing stuff. I've considered changing docs to use the git managed stuff, but I think there's an occasional problem depending on time/mars alignment/which foot you're standing on at the moment so #BetterSafeThanSorry

**Comment by dinger1986** [comment]
Yeah so checked (with Wh1te) and it is meant to be there and updates with every update.sh.

In theory when setting up backups you want to check you have the newest backup script it's good practice hence the docs being different.

So docs will stay the same but feel free to point to that one. (Need to chmod +x it)
